data_steward/retraction/retract_utils.py

```diff
@@ -79,6 +79,51 @@ def get_src_id(mapping_type):
     return src_id
 
 
+def get_datasets_list(project_id, dataset_ids):
+    # initialize list of all datasets in project
+    all_dataset_ids = [
+        dataset.dataset_id for dataset in bq.list_datasets(project_id)
+    ]
+
+    if dataset_ids == 'all_datasets':
+        dataset_ids = all_dataset_ids
+    elif dataset_ids == 'none':
+        dataset_ids = []
+    else:
+        dataset_ids = dataset_ids.split()
+        # only consider datasets that exist in the project
+        dataset_ids = [
+            dataset_id for dataset_id in dataset_ids
+            if dataset_id in all_dataset_ids
+        ]
+
+    logging.info('Found datasets to retract from: %s' % ', '.join(dataset_ids))
+    # retract from latest datasets first
+    dataset_ids.sort(reverse=True)
```

Comment:
This is correct, our naming convention has changed since the last update to this. Do you think it would be useful to retract from the latest datasets first? If not I will remove the comment and sorting. If yes I can split by '_' and grab the release tag, sorting by that instead, using a lambda function for the sort.

Reply:
Splitting by `_` would still leave you with this issue because then you'd be sorting by `C2021q1r1', `R2020q4r1', and `2021q1r1`.  I wouldn't try to sort it yet.  Because, even if you did manage to sort by `YYYYq#r#` only, you'd then have to potentially weigh the importance of the rdr, ehr, combined, registered, and controlled datasets.  The sorting can be an enhancement to this later.  As of now, I'd just try to get the expected values.